src/gmpints.c

```diff
@@ -375,6 +375,31 @@ Obj ObjInt_UInt( UInt i )
   }
 }
 
+Obj ObjInt_Int8( Int8 i )
+{
+  int neg = 0;
+  if(i < 0) {
+    neg = 1;
+    i = -i;
+  }
```

Comment:
Sorry, I am (too) late for reviewing, but: This code is incorrect for the smalles negative 64 bit integer.

The code is also quite inefficient; for converting a single integer, we may created at least four temporary integers in the GASMAN heap which later get garbage collected. This should affect Boehm more than it affects GASMAN.

Granted, this function is not used much right now, but I think for such a low-level function, we should implement it properly.